### PR TITLE
Some asm! diagnostic adjustments and a papercut fix

### PR DESCRIPTION
**Pull request by oli-obk**
Best reviewed commit by commit.

We forgot a `normalize` call in intrinsic checking, causing us to allow literal integers, but not named constants containing that literal. This can in theory affect stable code, but only if libstd contains a stable SIMD type that has an array length that is a named constant. I'd assume we'd have noticed by now due to asm! rejecting those outright.

The error message left me scratching my head for a bit, so I added some extra information to the diagnostic, too.

### CODE REVIEW

File: tests/ui/asm/named_const_simd_vec_len.stderr
Comment (jieyouxu) [inline tests/ui/asm/named_const_simd_vec_len.stderr:1]: Looks stray

**Comment by jieyouxu** [comment]
r? jieyouxu

**Review by jieyouxu — APPROVED** [review]
Thanks, this looks good to me. The diagnostics indeed look significantly nicer!

**Comment by fmease** [comment]
https://github.com/rust-lang/rust/pull/134097#issuecomment-2530610683
@bors r-

**Comment by fmease** [comment]
https://github.com/rust-lang/rust/pull/134133#issuecomment-2532629012
@bors r-

**Comment by fmease** [comment]
Other test needs to be limited, too.

**Comment by jieyouxu** [comment]
In the second commit, there's an empty stderr file as the test became `run-pass`.
@bors r-

**Comment by oli-obk** [comment]
What the heck xD I blessed all ui tests in all commits...

**Comment by jieyouxu** [comment]
> What the heck xD I blessed all ui tests in all commits...

Hm, this might be a compiletest bug, where if u had a previous stderr from check-fail then u change it to check-pass it might write empty stderr but keep the file... I'll open an issue later.

**Review by jieyouxu — APPROVED** [review]
Thanks!

**Pull request merged**